modin/experimental/batch/test/test_pipeline.py

```diff
@@ -0,0 +1,322 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import modin.pandas as pd
+import warnings
+import numpy as np
+from modin.config import Engine, NPartitions
+from modin.distributed.dataframe.pandas.partitions import from_partitions
+from modin.pandas.test.utils import df_equals
+
+@pytest.mark.skipif(
+    Engine.get() != "Ray",
+    reason="Only Ray supports the Batch Pipeline API",
+)
+def test_pipeline_simple():
```

Comment:
Could you document what the pipeline test here is supposed to test? I did not see any correctness checks until I dug deeper into the code because. (If you could, please also document the rest of these).

Also, can you factor out the `with pytest.warns` into a separate unit test? It makes this one feel very messy.

Reply:
Sure!